lib/pulsar/broker.ex

```diff
@@ -565,6 +574,28 @@ defmodule Pulsar.Broker do
     end
   end
 
+  def connected({:call, from}, {:publish_batch_message, command_send, batch_metadata, messages, compression}, broker) do
+    %__MODULE__{socket_module: mod, socket: socket} = broker
+
+    encoded_message = Pulsar.Protocol.encode_batch_message(command_send, batch_metadata, messages, compression)
```

Comment:
I suggest moving this to the producer. Otherwise, the broker will become the bottleneck.